src/main/java/org/spongepowered/eventimplgen/EventInterfaceProcessor.java

```diff
@@ -61,6 +61,6 @@ public void process(CtInterface<?> event) {
     }
 
     public static boolean shouldGenerate(CtInterface<?> candidate) {
-        return candidate.getNestedTypes().isEmpty() || EventImplGenTask.getAnnotation(candidate, "org.spongepowered.api.util.annotation.eventgen.GenerateFactoryMethod") != null;
+        return (candidate.getNestedTypes().isEmpty() && EventImplGenTask.getAnnotation(candidate, "org.spongepowered.api.util.annotation.eventgen.AbstractEvent") == null) || EventImplGenTask.getAnnotation(candidate, "org.spongepowered.api.util.annotation.eventgen.GenerateFactoryMethod") != null;
```

Comment:
Can you either split this into multiple if statements or at least add some line breaks here? There is way too much scrolling required here and it doesn't quite make it easier to understand :)